### ISSUE TITLE
RLS: 1.2.1

### ISSUE THREAD

**Issue by simonjayhawkins** [pandas-dev/pandas#38721]
Tracking issue for the 1.2.1 release.

https://github.com/pandas-dev/pandas/milestone/81

Currently scheduled for January 18, 2021 (date flexible on severity of regressions)

List of open regressions: https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression

**Comment by simonjayhawkins** [comment]
current status: 8 open regressions milestoned 1.2.1, 2 with open PRs

https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression+milestone%3A1.2.1

(#38752 is a actually a regression in 1.1 but is milestoned 1.2.1)

**Comment by simonjayhawkins** [comment]
status update...

a week away from scheduled release. 9 open regressions milestoned 1.2.1, 1 with open PR #38980

https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression+milestone%3A1.2.1

#38969 is effectively a duplicate of #38947

#38851 maybe tricky to fix so maybe best to revert cleanup for 1.2.1. https://github.com/pandas-dev/pandas/issues/38851#issuecomment-753247601

#38946 is maybe not a priority so could remove from milestone https://github.com/pandas-dev/pandas/issues/38946#issuecomment-754044905

**Comment by jreback** [comment]
ok, can certainly try on these. but let's just do a time release. whatever we have is fine.

**Comment by simonjayhawkins** [comment]
> ok, can certainly try on these. but let's just do a time release. whatever we have is fine.

yep, certainly on the patch releases (except the final in the series) a calendar based release schedule is fine.

**Comment by simonjayhawkins** [comment]
we've got the blocker label on #38787, should maybe discuss this on this week's dev call. cc @jorisvandenbossche @mroeschke

**Comment by simonjayhawkins** [comment]
although need not block, can revert backport, tag release, and re-apply backport

**Comment by jorisvandenbossche** [comment]
> we've got the blocker label on #38787, should maybe discuss this on this week's dev call.

Yes, that would be a good topic to discuss (although don't know if Matt will be there). But I will try to look again at the issue and post my thoughts there.

**Comment by mroeschke** [comment]
I believe I can make the dev call this Wednesday to discuss #38787

**Comment by simonjayhawkins** [comment]
in the dev meeting notes for the release we have

> Maybe revert plotting PR

@jorisvandenbossche I believe you suggested this.

I think that we should not yet revert until have investigated the issues further. 1.2.2 is scheduled for February 07, 2021 (3 weeks)

current status: 

https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression+milestone%3A1.2.1

9 open regressions milestoned 1.2.1

5 linked to the visualization issue #39126, #38969, #38947, #38865 and #38736

#39184 requires further discussion/agreement on preferred course of action

#39115 and #39168 have open PRs

#38946 is not a priority for 1.2.1

**Comment by simonjayhawkins** [comment]
> I think that we should not yet revert until have investigated the issues further. 1.2.2 is scheduled for February 07, 2021 (3 weeks)

from my limited understanding of the changes in that PR, I'm leaning towards the revert option.

If we do revert (for 1.2.1), should also have tests to prevent future regression. I'll open a PR for the revert, just in case, but won't be adding regression tests, so would need to leave the issues open with a 'needs tests' label.

**Comment by jorisvandenbossche** [comment]
I think that would be OK to just do the revert now, and add tests in follow-up PRs.

**Comment by simonjayhawkins** [comment]
4 open PRs milestoned 1.2.1. These need to be merged or moved off milestone before we start the release process

https://github.com/pandas-dev/pandas/pulls?q=is%3Apr+is%3Aopen+milestone%3A1.2.1

**Comment by jorisvandenbossche** [comment]
https://github.com/pandas-dev/pandas/pull/39239 is obviously a last minute change, but if people agree on the deprecation, I think we should try to include it. I *think* my patch is relatively safe, since converting the input to numpy arrays (what I am doing now manually as fallback) is what happened before adding `DataFrame.__array_ufunc__` as well. 

If I can get some review on the PR, we should be able to merge it in the afternoon

**Comment by jreback** [comment]
let's move them all off 1.2.1

last minute changes are always a bad idea

**Comment by simonjayhawkins** [comment]
final pre-release checklist

- [ ]  https://github.com/MacPython/pandas-wheels/pull/122
- [ ]  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
- [ ]  release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.1.html
- [ ]  ci checks https://github.com/pandas-dev/pandas/commits/1.2.x
- [ ]  no open issues/PRs (except this one) https://github.com/pandas-dev/pandas/milestone/81

**Comment by jorisvandenbossche** [comment]
@simonjayhawkins do you have time to give https://github.com/pandas-dev/pandas/pull/39239 a review? (Tom mentioned before he was OK with the change if we would be able to do it soon: https://github.com/pandas-dev/pandas/issues/39184#issuecomment-760975753)

**Comment by simonjayhawkins** [comment]
I've given it a quick once over, I'll look again in more detail.

**Comment by jorisvandenbossche** [comment]
@simonjayhawkins I moved the excel regressions to the 1.2.1 milestone. 

So with that, the queue is clear, and only waiting on CI to pass on the 1.2.x branch I think (I merged the backport)

**Comment by simonjayhawkins** [comment]
final pre-release checklist

- [ ]  https://github.com/MacPython/pandas-wheels/pull/122
- [ ]  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
- [ ]  release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.1.html
- [ ]  ci checks https://github.com/pandas-dev/pandas/commits/1.2.x
- [ ]  no open issues/PRs (except this one) https://github.com/pandas-dev/pandas/milestone/81

**Comment by simonjayhawkins** [comment]
starting release now.

- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag https://github.com/pandas-dev/pandas/tags
- [ ] open PR to build wheels https://github.com/MacPython/pandas-wheels/pull/XXX
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.2.1/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release - https://github.com/pandas-dev/pandas/releases
- [ ] upload wheels to PyPI https://pypi.org/project/pandas/
- [ ] change link for stable docs https://pandas.pydata.org/docs/
- [ ] conda-forge bot picked up on release https://github.com/conda-forge/pandas-feedstock/pulls
- [ ] available on conda-forge https://anaconda.org/conda-forge/pandas/files
- [ ] announce

**Comment by jorisvandenbossche** [comment]
Thanks for the release @simonjayhawkins !

**Comment by jreback** [comment]
thanks @simonjayhawkins !
